### PR TITLE
PromtInteraction and delete broadcast stuff

### PR DESCRIPTION
**Pull request by Donahuec**
This pull request creates a sub-class called 'PromptInteraction,' that contains the prompt information, and then subclasses  most interactions off of this instead, so that prompts are not required for  annotations. 

Also deleted broadcast classes.

### CODE REVIEW

**Review by grahamearley — APPROVED** [review]
Looks great!

Here's a request that you don't need to add in this PR, but it might be easy and would be nice: could you make it so that if a prompt string is blank, the GUI doesn't draw?

(if you don't want to, I'm fine merging this PR as is)

File: Project/Assets/Prairie/Framework/Script/Interaction/Prompt.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public void SetDefaultPrompt()
 	{
 		string prompt = "";
 		GameObject source = this.gameObject;
-		foreach (Interaction i in source.GetComponents<Interaction> ())
+		foreach (PromptInteraction i in source.GetComponents<Interaction> ())
```
Comment (cpimhoff) [inline Project/Assets/Prairie/Framework/Script/Interaction/Prompt.cs]: Minor Bug: should be `source.GetComponents<PromptInteraction> ()`

File: Project/Assets/Prairie/Framework/Script/Interaction/PromptInteraction.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using UnityEngine;
+using System.Collections;
+
+[RequireComponent(typeof(Prompt))]
+public abstract class PromptInteraction : Interaction
```
Comment (cpimhoff) [inline Project/Assets/Prairie/Framework/Script/Interaction/PromptInteraction.cs:5]: Slick.

**Review by cpimhoff — CHANGES_REQUESTED** [review]
I highlighted a minor bug - otherwise legit.

File: Project/Assets/Prairie/Framework/Script/Interaction/Prompt.cs (hunk 1 above)
Comment (Donahuec) [reply Project/Assets/Prairie/Framework/Script/Interaction/Prompt.cs]: Fixed!

**Comment by Donahuec** [comment]
Fixed bug, and also made it so that prompt wont draw i an existing prompt object is null or whitespace

**Review by cpimhoff — APPROVED** [review]

**Pull request merged**